include/tvm/relay/doc.h

```diff
@@ -0,0 +1,435 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file doc.h
+ * \brief A pretty printer DSL for constructing (Doc) and formatting (RDoc) documents.
+ *        It is based heavily on Philip Wadler's "A prettier printer."
+ *        See https://homepages.inf.ed.ac.uk/wadler/papers/prettier/prettier.pdf
+ *        for more details.
+ *
+ * Since the original paper uses call by value for efficiency, everything doc function is maximally lazy.
+ * You can probably yank speed by doing strict analysis and removing some Lazy (if this is bottleneck).
+ */
+#ifndef TVM_RELAY_DOC_H_
+#define TVM_RELAY_DOC_H_
+
+#include <map>
+#include <utility>
+#include <string>
+#include <functional>
+#include <vector>
+#include <memory>
+#include <ostream>
+#include "error.h"
+
+namespace tvm {
+namespace relay {
+
+struct DocError : Error {
```

Comment:
This part of the code need proper documentation on the data structure and member functions

Reply:
It would also be helpful to give a few examples on how it can be used(sequence of input, example output), currently the data structure do not speak for themselves